File: app/src/main/java/fr/nihilus/xenobladechronicles/MyApp.java
package fr.nihilus.xenobladechronicles;

import android.app.Activity;
import android.app.Application;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import fr.nihilus.xenobladechronicles.di.DaggerAppComponent;

public class MyApp extends Application implements HasActivityInjector {
    @Inject DispatchingAndroidInjector<Activity> dispatchingActivityInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        DaggerAppComponent.builder()
                .application(this)
                .build()
                .inject(this);
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return dispatchingActivityInjector;
    }
}

File: app/src/main/java/fr/nihilus/xenobladechronicles/MyApp.kt
package fr.nihilus.xenobladechronicles

import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import fr.nihilus.xenobladechronicles.di.DaggerAppComponent

class MyApp : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.factory().create(this)
    }
}
